### ISSUE TITLE
Feature request: no-floating-locators

### ISSUE THREAD

**Issue by mikedidomizio** [playwright-community/eslint-plugin-playwright#395]
In Testing Library, [querying](https://testing-library.com/docs/queries/about) (getBy*) will throw an error if no elements match.  Preventing the continuation of a test.
In Playwright, [locators](https://playwright.dev/docs/locators#introduction) although visually similar, behave differently, and will not fail written the same way as Testing Library.  The test continues.

The differences may not be clear to some people, and can result in flakiness due to not understanding the differences between the two testing frameworks.

Some proper usages:

```tsx
// variable assignment to be used later
const btn = page.getByRole('button', { name: 'Sign in' })
```

```tsx
// a promise (user action) that is awaited
await page.getByRole('button', { name: 'Sign in' }).click()
```

```tsx
// passing the result of the locator into an assertion
await expect(page.getByRole('button', { name: 'Sign in' })).toBeVisible()
```

Invalid usages:

```tsx
page.getByRole('button', { name: 'Sign in' })

// continues when developer expected test to wait for 'Sign in' to be visible/attached/etc. similar to Testing Library.
```

My thinking is a rule (named similar to [no-floating-promises](https://typescript-eslint.io/rules/no-floating-promises/)) that requires the developer to use the locator in one of the following proper ways and can't "float".  I can't think of a scenario where the invalid usage by itself makes sense (but let me know if I'm wrong).
 
- I've looked through the existing rules of this plugin and I didn't see anything that covered this.
- I've looked through Playwright options for something that might help and didn't find anything.
- I've tried to think of other cases where the invalid usage is acceptable.

**Comment by mskelton** [comment]
The idea is good, but I would rename it to `no-unused-locators`
